File: src/Services/Book/Book.Application/DTOs/Book/ResponseDTOs/BookResponseDto.cs
using Book.Application.DTOs.Author.ResponseDTOs;
using Book.Application.DTOs.Language.ResponseDTOs;
using Book.Application.DTOs.Subject.ResponseDTOs;

namespace Book.Application.DTOs.Book.ResponseDTOs;

public record BookResponseDto(
    int Id,
    string? OpenLibraryKey,
    string Title,
    int EditionCount,
    int PublicationYear,
    double AverageRating,
    string? ImageUrl,
    IEnumerable<AuthorResponseDto> Authors,
    IEnumerable<LanguageResponseDto> Languages,
    IEnumerable<SubjectResponseDto> Subjects);

File: src/Client/Angular.Client/src/app/models/bookResponseDto.ts
import {AuthorResponseDto} from "./authorResponseDto";
import {LanguageResponseDto} from "./languageResponseDto";
import {SubjectResponseDto} from "./subjectResponseDto";

export interface BookResponseDtoBook {
  id: number;
  openLibraryKey?: string | null;
  title: string;
  editionCount: number;
  publicationYear: number;
  averageRating: number;
  imageUrl?: string | null;
  authors: AuthorResponseDto[];
  languages: LanguageResponseDto[];
  subjects: SubjectResponseDto[];
}
